Assets/Scripts/Interactables/Dialogue/DialogueInteractable.cs

```diff
@@ -1,92 +1,147 @@
-{
-    public Conversation conversation;
-
-    private SpeakerUI speakerUIBirdie;
-    private SpeakerUI speakerUINPC;
-
-    protected bool isConversing = false;
-
-    private int activeLineIndex = 0;
-
-    public override void OnInteract()  
-    {
-        if (!isConversing)
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DialogueInteractable : Interactable
+{
+    public Conversation conversation;
+
+    private SpeakerUI speakerUIBirdie;
+    private SpeakerUI speakerUINPC;
+
+    protected bool isConversing = false;
+    private int activeLineIndex = 0;
+
+    protected bool autoPlaying = false;
+
+    public float waitLineTime = Constants.WAIT_TIME_CONVO_LINE;
+
+
+    protected override void Start()
+    {
+        base.Start();
+        speakerUINPC = Utils.GetRequiredComponentInChildren<SpeakerUI>(this);
+    }
+
+    public override void OnInteract()
+    {
+        speakerUIBirdie = Utils.GetRequiredComponentInChildren<SpeakerUI>(player);
+        if(!autoPlaying)
         {
-            isConversing = true;
+            if (!isConversing)
+            {
+                isConversing = true;
 
-            // Freeze player when conversing
-            player.GetComponent<PlayerController>().CanMove = false;
+                // Freeze player when conversing
+                player.GetComponent<PlayerController>().CanMove = false;
 
-            speakerUIBirdie = Utils.GetRequiredComponentInChildren<SpeakerUI>(player);
-            speakerUINPC = Utils.GetRequiredComponentInChildren<SpeakerUI>(this);
+                speakerUIBirdie = Utils.GetRequiredComponentInChildren<SpeakerUI>(player);
+                speakerUINPC = Utils.GetRequiredComponentInChildren<SpeakerUI>(this);
 
-            HideInteractUI();
-            AdvanceConversation();
+                HideInteractUI();
+                AdvanceConversation();
 
-            base.OnInteract();
-        }
+                base.OnInteract();
+            }
+            else
+            {
+                AdvanceConversation();
+            }
+        }
         else
         {
-            AdvanceConversation();
-        }
-    }
-
-    void AdvanceConversation() {
-        if (activeLineIndex < conversation.lines.Length)
-        {
-            //If there is still conversation left
-            DisplayLine();
-            activeLineIndex += 1;
-        }
-        else
-        {
-            //Once the conversation is over 
-            //what happens 
-            speakerUIBirdie.Hide();
-            speakerUINPC.Hide();
-            activeLineIndex = 0;
-
-            isConversing = false;
+            StartCoroutine(AutoplayConversation()); ;
+        }
+    }
+
+    protected void TriggerAutoplay()
+    {
+        autoPlaying = true;
+    }
+
+    protected virtual void OnAutoplayComplete()
+    {
+        autoPlaying = false;
+    }
+
+    private bool ContinueConversation()
+    {
+        //DisplayLine();
```

Comment:
Don't need this commented out line anymore?

Reply:
Actually, don't we need this line and the index increase in this function, so we cut down some logic?

```
DisplayLine();
activeLineIndex += 1;
```

Reply:
Yeah this was the issue that I wanted to talk to you about because by displaying the line like this and then doing `activeLineIndex += 1` it would not display the last line because it would go straight into ending the conversation.

Reply:
So right now I did a temporary solution which was revert it back to how it worked before.

Reply:
I'm not sure I see how it'll end the conversation early like that, I may be missing something though.

If you did like,
```
DisplayLine();
activeLineIndex += 1;
return activeLineIndex - 1 < conversation.lines.Length;
```
I imagine it'll show all lines (maybe you didn't try the `- 1` though?)

Reply:
yeah I didnt try the -1 LOL

Reply:
I'll try it now and get back to you :)

Reply:
Ok nvm I did try it and it started giving index out of range because it would run another iteration. The reason why it ended the conversation early is because it would display the last line in `ContinueConversation` but then it would return false and then immediately proceed to `EndConversation`. So technically it did display the last line but we just can't see it.

Reply:
Weird... we'll look into it after I guess let;s not worry too much about it now I guess. 
We'll just focus on finishing up alpha aha